rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/api/Coordinator.java

```diff
@@ -238,13 +239,34 @@ public Response startLRA(
                     return Response.status(response.getStatus()).build();
                 }
             } else {
-                ResponseHolder resp = new RequestBuilder(parentLRAUrl)
+                LRALogger.i18NLogger.error_parentLRANotFound(parentLRAUrl);
+
+                Client client = null;
+                Response response = null;
+
+                try {
+                    client = ClientBuilder.newClient();
+                    response = client.target(parentLRAUrl)
                         .request()
-                        .async(PARTICIPANT_TIMEOUT, TimeUnit.SECONDS)
-                        .put(compensatorUrl, MediaType.TEXT_PLAIN);
-                if (resp.getStatus() != Response.Status.OK.getStatusCode()) {
-                    return Response.status(resp.getStatus()).build(); // TODO include reason
+                        .async()
+                        .put(Entity.text(compensatorUrl))
+                        .get(PARTICIPANT_TIMEOUT, TimeUnit.SECONDS);
+
+                    if (response.getStatus() != Response.Status.OK.getStatusCode()) {
+                        String errMessage = String.format("The coordinator at %s returned an unexpected response: %d",
+                            parentLRAUrl, response.getStatus());
+                        return Response.status(response.getStatus()).entity(errMessage).build();
+                    }
+                } catch (Exception e) {
+                    LRALogger.logger.errorf("Cannot contact the LRA Coordinator at %s", parentLRA);
+                    return Response.status(PRECONDITION_FAILED).entity(e.getMessage()).build();
```

Comment:
This error code seems a bit random. The method signature is already handling it (ie it throws WebApplicationException). Also we don't need to log anything here (see my previous comments about logging errors). If you really did want to log it at trace level then just re-throw the WebApplicationException.

Reply:
+1